### ISSUE TITLE
SINGULARITY_TMPDIR=/var/singularity/tmp  singularity build not using the env variable, but failing to /tmp with not enough space

### ISSUE THREAD

**Issue by titansmc** [apptainer/singularity#2305]
## Version of Singularity:

latest pull from master

### Expected behavior

build image in specific tmpdir

### Actual behavior

````
SINGULARITY_TMPDIR=/var/singularity/tmp  singularity --verbose  build ngless Singularity
WARNING: Authentication token file not found : Only pulls of public images will succeed
INFO:    Starting build...
Getting image source signatures
Skipping fetch of repeat blob sha256:f9f73d801f0558b085ffa505240a065319269c4cefbe9c2e60103d58761edfa8
Copying config sha256:a364696682bd9b114799e2764041ee44f7e0573d263e899641bf66c642615f2d
 857 B / 857 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
FATAL:   While performing build: conveyor failed to get: Error writing blob: write /tmp/sbuild-256120010/oci-put-blob738714252: no space left on device

````

### Steps to reproduce behavior

Set diferent path like:
  SINGULARITY_TMPDIR=/var/singularity/tmp  singularity --verbose  build ngless Singularity

**Comment by jmstover** [comment]
@titansmc 
I believe that you use: `TMPDIR=/var/singularity/tmp singularity build [...]`

Because of the change to Go, we're using a different code path than we previously did, so just setting `TMPDIR` works.

**Comment by jscook2345** [comment]
TMPDIR is the correct option to use here @titansmc . Please let us know if it does not work for you!

Side note, this should be documented as deprecated or put in the changelog as removed.

**Comment by titansmc** [comment]
Works like a charm. Thank.

**Comment by jscook2345** [comment]
@bauerm97 

Where should this go in the changelog? A new section on removed environment variables?

**Comment by ikaneshiro** [comment]
Hey @titansmc!
This is actually a regression, it should be `SINGULARITY_TMPDIR` and this will be fixed by PR #2396
